playpen/repo_level_awareness/agents/compiler_agent.py

```diff
@@ -0,0 +1,138 @@
+from playpen.repo_level_awareness.api import Task, TaskResult, Agent
+from playpen.repo_level_awareness.maven_validator import MavenCompilerError
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_core.messages import HumanMessage, SystemMessage, BaseMessage
+from jinja2 import Template
+import asyncio
+from dataclasses import dataclass
+from pathlib import Path
+
+@dataclass
+class MavenCompilerLLMResponse():
+    reasoning: str
+    java_file: str
+    addional_information: str
+
+
+class MavenCompilerAgent(Agent):
+    """This agent is reponsible for taking a set of maven compiler issues and solving.
+     
+    For a given file it will asking LLM's for the changes that are needed for the at whole file
+    returning the results.
+    """
+
+    system_message=SystemMessage(content="""
+    I will give you compiler errors and the offending line of code, and you will need to use the file to determine how to fix them. You should only use compiler errors to determine what to fix.
+
+    Make sure that the references to any changed types are kept.
+
+    You must reason through the required changes and rewrite the Java file to make it compile. 
+
+    You will then provide an step-by-step explaination of the changes required tso that someone could recreate it in a similar situation. 
+    """)
+
+    chat_message_template=Template("""
+    [INST]
+    ## Compile Errors
+    {{compile_errors}}
+
+    ## Input File
+    {{src_file_contents}}
+
+
+    # Ouput Instructions 
+    Structure your output in Markdown format such as:
+
+    ## Updated Java File
+    Rewrite the java file here
+
+    ## Reasoning 
+    Write the step by step reasoning in this markdown section. If you are unsure of a step or reasoning, clearly state you are unsure and why. 
+
+    ## Addition Infomation (optional)
```

Comment:
Nit: Should this be "Additional"? Same for the parse_llm_response function below.

Reply:
updated